### ISSUE TITLE
Make it work with NC 26

### ISSUE THREAD

**Issue by MrLoverLoverMMMM** [nextcloud/cms_pico#236]
Hello!

It would be nice if it would work on NC 26 as well.
The Admin Page like Themes,... is not working there anymore.

best regards

Patrick

**Comment by Uatschitchun** [comment]
This would be NC27 now...

**Comment by EdG973GCS** [comment]
OK for hacky workaround.
But PICO CMS users would appreciate an update for NC 26 and above.
It is so stressful to get our pico pages not working after each new NC version deployment.

**Comment by MrLoverLoverMMMM** [comment]
I agree

**Comment by PhrozenByte** [comment]
Pull request on the [`cms_pico-1.0`](https://github.com/nextcloud/cms_pico/tree/cms_pico-1.0) branch is welcome :+1:

**Comment by jkraffthha** [comment]
OK, fixed.
To get it working, you have to modify the file /apps/cms_pico/js/admin.js and change the line
.tooltip('dispose')
to
.tooltip('destroy')

**Comment by jkraffthha** [comment]
> Pull request on the [`cms_pico-1.0`](https://github.com/nextcloud/cms_pico/tree/cms_pico-1.0) branch is welcome 👍

Fix is on the way, I've made a pull request.

**Comment by khassad** [comment]
Hi, any news on NC 26+ compatibility ?

**Comment by DivineDominion** [comment]
@PhrozenByte Is this still an issue or can this be closed?

**Comment by jkraffthha** [comment]
> @PhrozenByte Is this still an issue or can this be closed?

My fix has been commited to the 1.0 branch, I don't know what's missing for it to get to the Nextcloud App Store...

**Comment by DivineDominion** [comment]
The `cms_pico-1.0` branch? https://github.com/nextcloud/cms_pico/tree/cms_pico-1.0

I guess this needs to be merged into the main branch before the App Store picks this up at the very least. 🤔

**Comment by horga83** [comment]
Any movement about getting this into the NC27 app store?

**Comment by StCyr** [comment]
@DivineDominion, @jkraffthha,

I don't find the PR, where is it?

**Comment by jkraffthha** [comment]
> @DivineDominion, @jkraffthha,
> 
> I don't find the PR, where is it?

Here: https://github.com/nextcloud/cms_pico/pull/242

**Comment by miguelsan** [comment]
> My fix has been commited to the 1.0 branch, I don't know what's missing for it to get to the Nextcloud App Store...

Well, now we have an [answer](https://github.com/nextcloud/cms_pico/pull/242#issuecomment-1946072223) to that:

> [...] run a manual app install (see [`README.md`](https://github.com/nextcloud/cms_pico?tab=readme-ov-file#manually)) from the `cms_pico-1.0` branch (use `git clone -b cms_pico-1.0 https://github.com/nextcloud/cms_pico.git apps/cms_pico`) on a fresh Nextcloud instance running stable versions 26, 27 and 28 [...] including manual testing of all the app's basic functionality.

Can anybody confirm that it has been successfully tested in this way?

**Comment by miguelsan** [comment]
Now the need for a solution has become urgent!

At my install, I get the following under **Administration settings**:

```
Version
25.0.13

Update
  The version you are running is not maintained anymore. Please make sure to update to a supported version as soon as possible.
A new version is available: Nextcloud 26.0.13
1 app has no compatible version for this Nextcloud version available.
Apps missing compatible version
  Pico CMS
```

Any news on advancements?

**Comment by noci2012** [comment]
And currently there is Nextcloud 29.....
making 26 even obsolete.

In 29 the argument list for class creator is slightly different.

**Comment by noci2012** [comment]
That might be easy to someone already working with all the stuff, not so much NC actually.
Documentation missing, misdirecting inconsistent etc, 
I guess i have to take this as a statement of obsolesence and deprecation for cms_pico as one of the next apps fallen into the gutter.

**Comment by noci2012** [comment]
installing pico-1.0 from github  does work in 29.
There is a warning about using deprecated appinfo/app.php.

Just migrating the code there to lib/AppInfo/Application.php   gives other errors n some Http.php library.
Still the webpages do render.

**Comment by GVLLIFESTYLE** [comment]
With version 29. I have the issue, that my the settings-page of the user shows only 1 active site intsead of all of them + you can't change any settings anymore.
The error:

/appinfo/app.php is deprecated, use \OCP\AppFramework\Bootstrap\IBootstrap on the application class instead.

**Comment by fedora-mahdi** [comment]
Hi all !

This is a GREAT App, I would be a waste to see it unmaintained or tagged as obsolete

While the future updated release is still upcoming; the latest release dates back to Oct 22, 2022.

Can this release be still installed on any currently maintained Nextcloud version ? NC 28,29,30 ?

@GVLLIFESTYLE and @noci2012  : 
Could you please detail how to install it on a fresh latest NC instance with your workarounds and caveats ?
If you might provide a beginner step by step install , I would be PERFECT

Thanks !

**Comment by GVLLIFESTYLE** [comment]
I was not able to freshly install it.
I just had it already installed and reactivated it with each new version. The admin panel was defect, but the sites was still working. Since version 29,30,31 that though does not work at all anymore and I receive error 500 when I try to open a page.

**Comment by rusangl** [comment]
Some thoughts about Pico in Nextcloud 31. 

According to the logs it seems the basic problem is the Nextcloud team removed OCP\ILogger methods: https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/app_upgrade_guide/upgrade_to_31.html

OCP\ILogger should be replaced with Psr\Log\LoggerInterface.

As far as I understand, in general we have to 1) get the class 
 Psr\Log\LoggerInterface from Nextcloud in AppInfo.php file and 2) replace in all pico's php files string "use OCP\ILogger" with "use Psr\Log\LoggerInterface;" with some other small changes.

But I am not a programmist. Maybe somebody has enough knowledge and  can explain how to get the class  Psr\Log\LoggerInterface from Nextcloud correctly?

AI says I need to create container.php:

use OCP\AppFramework\App;
use Psr\Log\LoggerInterface;
$application = new App('cms_pico');
$container = $application->getContainer();
$container->registerService('LoggerInterface', function () use ($container) {
    return $container->query('ServerContainer')->getLogger();
});

And in AppInfo.php use:

public function __construct(array $urlParams = [], LoggerInterface $logger = null)
{                                                parent::__construct(self::APP_NAME, $urlParams);
    if ($logger !== null) {                                                $this->logger = $logger;
    } else {
        $this->logger = $container->query('ServerContainer')->getL>    }
}

But this causes 
"Could not resolve OCA\CMSPico\Service\Psr\Log\LoggerInterface! Class "OCA\CMSPico\Service\Psr\Log\LoggerInterface" does not exist","

**Comment by StCyr** [comment]
Hello,

I've fixed pico CMS on my Nextcloud 32 by doing the following:

- replace all "use OCP\ILogger" by "use Psr\Log\LoggerInterface"
- in lib/Migration/MigrationTrait.php, update the $this->logger->log methods by their corresponding $this->logger->warning/info/debug/... equivalent
- in "lib/AppInfo/Application.php" add "include_once __DIR__ . '/../../vendor/autoload.php';"
- in "appinfo/info.xml" update max-version

I will see if I can send a MR now

**Comment by StCyr** [comment]
hmm looks like @PhrozenByte merged a change that does almost all that a few days ago 👍 

@PhrozenByte, can you explain why I had to add "include_once DIR . '/../../vendor/autoload.php';" line? (I copied this loading method from the Mail app after I got a message like "in lib/Pico.php cannot find \Pico") 

Maybe @ChristophWurst can tell us if it's really needed or not, ping Christopher?

**Comment by ChristophWurst** [comment]
Yes you need to require the custom composer autoloader: https://docs.nextcloud.com/server/latest/developer_manual/app_development/dependency_management.html#composer

**Comment by Julrob199** [comment]
Any update on it :)?

A fix would be really helpful 🙏

**Comment by KaeTuuN** [comment]
Pico and cms_pico for Nextcloud are dead! Please stop riding dead horses, if you are not interested in reanimating them!

See readme.md for further informations.
